src/main/java/org/cascadebot/cascadebot/commands/music/SkipCommand.java

```diff
@@ -0,0 +1,136 @@
+
+package org.cascadebot.cascadebot.commands.music;
+
+import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
+import net.dv8tion.jda.core.entities.Member;
+import org.cascadebot.cascadebot.UnicodeConstants;
+import org.cascadebot.cascadebot.commandmeta.Argument;
+import org.cascadebot.cascadebot.commandmeta.ArgumentType;
+import org.cascadebot.cascadebot.commandmeta.CommandContext;
+import org.cascadebot.cascadebot.commandmeta.ICommandMain;
+import org.cascadebot.cascadebot.commandmeta.Module;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+import org.cascadebot.cascadebot.utils.buttons.Button;
+import org.cascadebot.cascadebot.utils.votes.VoteButtonGroup;
+import org.cascadebot.cascadebot.utils.votes.VoteButtonGroupBuilder;
+import org.cascadebot.cascadebot.utils.votes.VoteMessageType;
+import org.cascadebot.cascadebot.utils.votes.VoteResult;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class SkipCommand implements ICommandMain {
+
+    private static Map<Long, VoteButtonGroup> voteMap = new HashMap<>();
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        AudioTrack track = context.getMusicPlayer().getPlayer().getPlayingTrack();
+        if (track == null) {
+            context.getTypedMessaging().replyDanger("I'm not playing anything!");
+            return;
+        }
+
+        if (context.getMusicPlayer().getTracks().peek() == null) {
+            context.getTypedMessaging().replyDanger("No song to skip to!");
+            return;
+        }
+
+        if (context.getArgs().length > 0) {
+            if (context.getArg(0).equalsIgnoreCase("force")) {
+                if (context.hasPermission("skip.force")) {
+                    context.getMusicPlayer().skip();
+                    context.getTypedMessaging().replySuccess("Forcefully skipped the song");
+                } else {
+                    context.getUIMessaging().sendPermissionError("skip.force");
+                }
+                return;
+            }
+        }
+
+        if (voteMap.containsKey(context.getGuild().getIdLong())) {
+            if (context.getArgs().length > 0) {
+                if (context.getArg(0).equalsIgnoreCase("yes")) {
+                    voteMap.get(context.getGuild().getIdLong()).addVote(sender.getUser(), UnicodeConstants.TICK);
+                    return;
+                } else if (context.getArg(0).equalsIgnoreCase("no")) {
+                    voteMap.get(context.getGuild().getIdLong()).addVote(sender.getUser(), UnicodeConstants.RED_CROSS);
+                    return;
+                }
+            }
+            voteMap.get(context.getGuild().getIdLong()).addVote(sender.getUser(), UnicodeConstants.TICK);
+            context.getTypedMessaging().replyWarning("A skip vote is already running, but we added your vote automatically");
+            return;
+        }
+
+        if (track.getUserData().equals(sender.getUser().getIdLong())) {
```

Comment:
This should be above the two if statements above it

Reply:
acctuly no because it causes issues with voting

Reply:
Why does it? If it their own song they should always be able to skip?

Reply:
they can still skip just they can also now do `;skip no` before if they did that it whould just skip because of argument order

Reply:
I think you'll agree allowing them to vote no is important

Reply:
I see what you're saying 👌